### ISSUE TITLE
Terraform v0.13.5 sum() function not working

### ISSUE THREAD

**Issue by robertomoutinho** [hashicorp/terraform#27468]
### Terraform Version
```
Terraform v0.13.5
```

### Terraform Configuration Files
I have a variables file with:
```terraform
variable "app_fargate_cpu" {
  description = "The amount of CPU for the main application"
  type        = number
  default     = 2048
}

variable "app_fargate_memory" {
  description = "The amount of RAM for the main application"
  type        = number
  default     = 4096
}

variable "sidecar_app_fargate_cpu" {
  description = "The amount of CPU for the sidecar application"
  type        = number
  default     = 1024
}

variable "sidecar_app_fargate_memory" {
  description = "The amount of RAM for the sidecar application"
  type        = number
  default     = 2048
}
```

A locals declaration to create the list of all app and sidecars that references the variables

```terraform
locals {

  total_fargate_cpu = tolist( 
                                 [
                                    tonumber(var.app_fargate_cpu), 
                                    tonumber(var.sidecar_app_fargate_cpu)
                                 ]
                               )

  total_fargate_memory = tolist( 
                                 [
                                    tonumber(var.app_fargate_memory), 
                                    tonumber(var.sidecar_app_fargate_memory)
                                 ]
                               )
                               
}

```

And the resource creation that references the local list

```terraform
  fargate_cpu           = sum(local.total_fargate_cpu)
  fargate_memory        = sum(local.total_fargate_memory)
```

### Expected Behavior

The sum of the variables to be passed to the resource creation as described at the docs: https://www.terraform.io/docs/configuration/functions/sum.html

```
sum takes a list or set of numbers 
```

and my error does say I'm passing a list of numbers with 2 values

```
local.total_fargate_memory is list of number with 2 elements
```

### Actual Behavior
```
Error: Invalid function argument

  on app.tf line 17, in module "onboarding_app":
  17:   fargate_memory        = sum(local.total_fargate_memory)
    |----------------
    | local.total_fargate_memory is list of number with 2 elements

Invalid value for "list" parameter: argument must be list, set, or tuple of
number values.
```

**Comment by jbardin** [comment]
Thanks @robertomoutinho 

I have confirmed this is fixed in our latest development release.

**Comment by robertomoutinho** [comment]
Hey @jbardin, Could you be more specific as to which version the fix is implemented?

**Comment by jbardin** [comment]
The latest development version is built from the `master` branch, and we have also released `0.15.0-alpha20210107` for preview. Thanks!
